hazelcast/src/main/java/com/hazelcast/config/EvictionConfig.java

```diff
@@ -66,19 +66,24 @@
      */
     public static final EvictionPolicy DEFAULT_EVICTION_POLICY = EvictionPolicy.LRU;
 
-    private static final int UNSET = -1;
-    private MaxSizePolicy maxSizePolicy = DEFAULT_MAX_SIZE_POLICY;
-    private EvictionPolicy evictionPolicy = DEFAULT_EVICTION_POLICY;
-    private String comparatorClassName;
-    private EvictionPolicyComparator comparator;
-    private int size = UNSET;
-    int defaultSize = DEFAULT_MAX_ENTRY_COUNT;
+    protected int size = DEFAULT_MAX_ENTRY_COUNT;
+    protected MaxSizePolicy maxSizePolicy = DEFAULT_MAX_SIZE_POLICY;
+    protected EvictionPolicy evictionPolicy = DEFAULT_EVICTION_POLICY;
+
+    protected String comparatorClassName;
+    protected EvictionPolicyComparator comparator;
+
+    /**
+     * Used by the {@link NearCacheConfigAccessor} to
+     * initialize the proper default value for on-heap maps.
+     */
+    boolean sizeConfigured;
```

Comment:
if `sizeConfigured` was set to `true`, after deserialization what keeps it still `true`?

Reply:
I have reverted the EvictionConfig behaviour. So it seems that setting sizeConfigured on deserialization was not necessary. For the sake of keeping it backward compatible, I will not touch this. Lets consider if that is necessary as a separate fix if needed.

Reply:
thanks, i see this is an existing issue, not introduced in this PR. Created an issue for it: https://github.com/hazelcast/hazelcast/issues/18354